src/main/webapp/app/course/plagiarism-cases/instructor-view/plagiarism-cases-instructor-view.component.html

```diff
@@ -0,0 +1,62 @@
+<h3 jhiTranslate="artemisApp.plagiarism.cases.plagiarismCases">Plagiarism Cases</h3>
+<h4 jhiTranslate="artemisApp.plagiarism.cases.pageSubtitle"></h4>
+
+<div *ngIf="plagiarismCases && plagiarismCases.length > 0; else noPlagiarismCases">
+    <div class="plagiarism-cases--toolbar row mb-3">
+        <div class="col-auto">
+            <button class="btn btn-primary" (click)="exportPlagiarismCases()">{{ 'artemisApp.plagiarism.cases.exportCsv' | artemisTranslate }}</button>
+        </div>
+    </div>
+    <div *ngFor="let exercise of exercisesWithPlagiarismCases; let i = index" class="card mb-2">
+        <div class="card-header">
+            <div class="row">
+                <div class="col-6">
+                    <h5 class="mb-0">
+                        <fa-icon *ngIf="exercise.type" [icon]="getIcon(exercise.type)"></fa-icon>
+                        {{ exercise.title }}
+                    </h5>
+                </div>
+                <div class="col-3">
+                    <jhi-progress-bar
+                        [denominator]="numberOfCases(groupedPlagiarismCases[exercise!.id!])"
+                        [numerator]="numberOfCasesWithPost(groupedPlagiarismCases[exercise!.id!])"
+                        [percentage]="(numberOfCasesWithPost(groupedPlagiarismCases[exercise!.id!]) / numberOfCases(groupedPlagiarismCases[exercise!.id!])) * 100 || 0"
```

Comment:
It may be better to extract this logic as a method in the component and call it here for conciseness, since there are multiple method calls, I guess Angular compiler won't be able to optimize and cache the result either way.

Reply:
Yes, good catch! I extracted it into a method now 👍🏼